### PR TITLE
fix(uzf): enforce surfdep < cell thickness

### PR DESCRIPTION
**Pull request by emorway-usgs**
Addresses #596

### CODE REVIEW

File: src/Model/GroundWaterFlow/gwf3uzf8.f90 [hunk 1]
```diff
@@ -2219,6 +2220,13 @@ subroutine read_cell_properties(this)
             'SURFDEP for uzf cell', i,                                           &
             'must be greater than 0 (specified value is', surfdep, ').'
           call store_error(errmsg)
+        endif
+        if(surfdep >= this%GWFTOP(i) - this%GWFBOT(i)) then
+          write(errmsg,'(a,1x,i0,1x,a)')                                        &
+            'SURFDEP for uzf cell', i,                                           &
+            'cannot be greater than the cell thickness.'
+          call store_error(errmsg)
+          surfdep = 0.2 * (this%GWFTOP(i) - this%GWFBOT(i))
```
Comment (langevin-usgs) [inline src/Model/GroundWaterFlow/gwf3uzf8.f90:2229]: Is this statement still needed?

File: src/Model/GroundWaterFlow/gwf3uzf8.f90 [hunk 2]
```diff
@@ -23,8 +23,9 @@ module UzfModule
   use ObserveModule, only: ObserveType
   use ObsModule, only: ObsType
   use InputOutputModule, only: URWORD
-  use SimVariablesModule, only: errmsg
-  use SimModule, only: count_errors, store_error, ustop, store_error_unit
+  use SimVariablesModule, only: errmsg, warnmsg
+  use SimModule, only: count_errors, store_error, ustop, store_error_unit,      &
+                       store_warning
```
Comment (langevin-usgs) [inline src/Model/GroundWaterFlow/gwf3uzf8.f90:28]: Probably don't need store_warning any more

Comment (langevin-usgs) [reply src/Model/GroundWaterFlow/gwf3uzf8.f90:28]: or warnmsg

Comment (emorway-usgs) [reply src/Model/GroundWaterFlow/gwf3uzf8.f90]: Have removed the adjustments that included warnmsg

File: src/Model/GroundWaterFlow/gwf3uzf8.f90 (hunk 1 above)
Comment (emorway-usgs) [reply src/Model/GroundWaterFlow/gwf3uzf8.f90]: Have removed the line of code that modified surfdep

**Comment by emorway-usgs** [comment]
This new check is accompanied by a new autotest

**Pull request merged**